ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto

```diff
@@ -602,6 +602,7 @@ message MissingInputState {
 }
 message InvalidParticipantState {
   string details = 1;
+  map<string, string> metadata = 2;
```

Comment:
This is more or less what we discussed with @SamirTalwar-DA today. I think this is the smallest change to propagate more information from the `Err` without introducing more `Rejection`s. @SamirTalwar-DA, @miklos-da WDYT? Is it worth doing?

Reply:
I think it's a good idea to introduce this first, and then break down the subclasses into their own error types in the future.

If you do it the other way around you have to do both at once.

Reply:
Agreed, this looks like a good approach.

Reply:
Do you have an idea about how the proto will evolve?

Reply:
To me, it makes sense to actually keep the `InvalidParticipantState` error, as this should be a really rare one. So, if you ask me, I don't think we're gonna change the errors soon.